### DISCUSSION TITLE
LFX Mentorship 2023 03-Sep-Nov Challenge - for #2690

### DISCUSSION THREAD

**Discussion by juntao** [WasmEdge/WasmEdge#2702] [Show and tell]
For those who want to apply for LFX mentorship for #2690, this is a pretest for the application.

1. Complete pre-test #2692 to demonstrate that you can write and build a basic Rust program.
2. Build and run a YOLO app in Python to detect and label objects on a static image.
3. Identify Python functions in the program that must be ported to Rust.

**Comment by h4shk4t** [comment]
Hello @juntao 

This is my submission for pretest 2702: https://github.com/h4shk4t/yolo-object-detection

I have added a README which explains how I have implemented YOLO in python and what are the possible ways and routes of porting the application to Rust. 

I would really appreciate suggestions and changes to improve the application.

Best Regards
Ashutosh

**Reply by juntao** [reply]
@h4shk4t Received. Thank you!

**Comment by aaryac33** [comment]
@juntao PFB my submission for pretest 2702:

https://github.com/aaryac33/yolo_tool

Regards,
Aarya

**Reply by alabulei1** [reply]
Hi @aaryac33,

 Could you please tell me the name when you apply for the LFX mentorship?

**Comment by Charles-Schleich** [comment]
Hi @juntao 

My submission for for this pretest is linked below.
https://github.com/Charles-Schleich/YOLO-object-detector

Best regards
Charles

**Reply by Charles-Schleich** [reply]
@juntao it looks like the CV+ Cover letter submission link https://mentorship.lfx.linuxfoundation.org/project/63adcf97-cd3e-4c96-a5fc-6076b5a5d511
is no longer taking applications.
The deadline on Secondstate's website was Aug 29th, 5:00 PM PDT. Is there a mistake perchance ?

**Reply by juntao** [reply]
Very cool. Thank you. So basically all the pre and post processing functions are OpenCV functions?

**Reply by juntao** [reply]
I will look into the submission issue

**Reply by Charles-Schleich** [reply]
Most of the pre and post processing functions look like they exist in OpenCV already.
The output tensor of the network does need a bit of processing and filtering before we can get the bounding boxes of the original input image. 
For the model I was using (yolov7_tiny) the output tensor had a shape `[1, 18900, 85]`
The function I implemented to process that tensor  `process_output` does make use of  numpy functions `squeeze`,`max`,`argmax`, as well `flatten` on `ndarrays`
So these would likely also need to be ported over to support the YOLO crate.
I'm also using a bunch of numpy-specific helper functions, `divide` `array` `copy` but these will be replaced by inbuilt Rust functions from Rust  potentially [nalgebra](https://nalgebra.org/), or even `opencvmini`.

**Reply by alabulei1** [reply]
Hi @Charles-Schleich ,

Could you please tell me the name when you apply for the LFX mentorship? I searched Schleich and found nothing.

**Reply by Charles-Schleich** [reply]
Hi @alabulei1 

I applied under the name `Charles Schleich`
I submitted my application on the 24th of August, and got a confirmation email for my application at Thu, Aug 24, 12:00 PM CEST.

**Reply by alabulei1** [reply]
Oh. I found it on the website. Could you please send me your resume via email? My email address is vivian dot xiage at gmail dot com. Thanks.

**Reply by Charles-Schleich** [reply]
@alabulei1 Sure thing, email has been sent.

**Comment by alabulei1** [comment]
> @juntao it looks like the CV+ Cover letter submission link https://mentorship.lfx.linuxfoundation.org/project/63adcf97-cd3e-4c96-a5fc-6076b5a5d511 is no longer taking applications. The deadline on Secondstate's website was Aug 29th, 5:00 PM PDT. Is there a mistake perchance ?

Hi @Charles-Schleich, I am sorry we have made a mistake on our web site. We are communicating with CNCF to extend the application deadline for you. Will let you know if I have any updates. Sorry again.

**Reply by Charles-Schleich** [reply]
@alabulei1 no problem at all. Looking forward to submitting my application.

**Comment by alabulei1** [comment]
Hi @Charles-Schleich, @h4shk4t, and @aaryac33,

Thank you for completing the pretest. If you have not yet submitted your application for this project, please do so before 5 PM PDT on August 25th. 

Apply here: https://mentorship.lfx.linuxfoundation.org/project/63adcf97-cd3e-4c96-a5fc-6076b5a5d511

**Comment by ShounakDeshmukh** [comment]
Hi @juntao and @alabulei1 
This is my submission for the pretest : [ShounakDeshmukh/Yolo-model-opencv](https://github.com/ShounakDeshmukh/Yolo-model-opencv)

I have added a README explaining how some of the functionality can be written in pure Rust

**Comment by aaryac33** [comment]
Hi,

I applied for the mentorship by the name Aarya Chaudhari and I applied on
13th/14th Aug.

Looking forward to hearing from you.

Regards,
Aarya

**Reply by alabulei1** [reply]
Yeah. I found your application. Thanks!

**Comment by aaryac33** [comment]
Sure!
I’ll send the resume in the next 2 hours.

Thank you.

Regards,
Aarya

**Comment by aaryac33** [comment]
Mailed the resume!

Thank you.

**Comment by ShounakDeshmukh** [comment]
Hi @alabulei1,
Have you gotten my application 
I applied on the 8th of Aug
but was unaware of the pre tasks
So apologies for any delays
Looking forward to hearing from you.

Regards,
Shounak

**Reply by alabulei1** [reply]
Yes, I found your application. Don't worry. So I didn't ask you here .
